plugins/lookup/vault_kv2_get.py

```diff
@@ -0,0 +1,230 @@
+    type: str
+    required: True
+  backend_mount_point:
+    default: kv
+  version:
+    description: Specifies the version to return. If not set the latest version is returned.
+    type: int
+'''
+
+EXAMPLES = r'''
+- name: Read a kv2 secret with the default mount point
+  ansible.builtin.set_fact:
+    response: "{{ lookup('community.hashi_vault.vault_kv2_get', 'hello', url='https://vault:8201') }}"
+  # equivalent API path is secret/data/hello
+
+- name: Display the results
+  ansible.builtin.debug:
+    msg:
+      - "Secret: {{ response.secret }}"
+      - "Data: {{ response.data }} (contains secret data & metadata in kv2)"
+      - "Metadata: {{ response.metadata }}"
+      - "Full response: {{ response.raw }}"
+      - "Value of key 'password' in the secret: {{ response.secret.password }}"
+
+- name: Read version 5 of a kv2 secret with a different mount point
+  ansible.builtin.set_fact:
+    response: "{{ lookup('community.hashi_vault.vault_kv2_get', 'hello', version=5, backend_mount_point='custom/kv2/mount', url='https://vault:8201') }}"
+  # equivalent API path is custom/kv2/mount/data/hello
+
+- name: Assert that the version returned is as expected
+  ansible.builtin.assert:
+    that:
+      - response.metadata.version == 5
+
+- name: Perform multiple kv2 reads with a single Vault login, showing the secrets
+  vars:
+    paths:
+      - hello
+      - my-secret/one
+      - my-secret/two
+  ansible.builtin.debug:
+    msg: "{{ lookup('community.hashi_vault.vault_kv2_get', *paths, auth_method='userpass', username=user, password=pwd)['secret'] }}"
+
+- name: Perform multiple kv2 reads with a single Vault login in a loop
+  vars:
+    paths:
+      - hello
+      - my-secret/one
+      - my-secret/two
+  ansible.builtin.debug:
+    msg: '{{ item }}'
+  loop: "{{ query('community.hashi_vault.vault_kv2_get', *paths, auth_method='userpass', username=user, password=pwd) }}"
+
+- name: Perform multiple kv2 reads with a single Vault login in a loop (via with_), display values only
+  vars:
+    ansible_hashi_vault_auth_method: userpass
+    ansible_hashi_vault_username: '{{ user }}'
+    ansible_hashi_vault_password: '{{ pwd }}'
+  ansible.builtin.debug:
+    msg: '{{ item.values() | list }}'
+  with_community.hashi_vault.vault_kv2_get:
+    - hello
+    - my-secret/one
+    - my-secret/two
+'''
+
+RETURN = r'''
+_raw:
+  description:
+    - The result of the read(s) against the given path(s).
+  type: list
+  elements: dict
+  contains:
+    raw:
+      description: The raw result of the read against the given path.
+      returned: success
+      type: dict
+      sample:
+        auth: null
+        data:
+          data:
+            Key1: value1
+            Key2: value2
+          metadata:
+            created_time: "2022-04-21T15:56:58.8525402Z"
+            custom_metadata": null
+            deletion_time": ""
+            destroyed": false
+            version": 2
```

Comment:
Super late, but are these missing opening quotes?

Reply:
I think it's more that they have closing quotes that shouldn't be there :-)

This is actually valid YAML (just checked), you get a dictionary where the keys end with `"`. Which is likely not what the API returns.

Reply:
yep, that was not intentional.. thanks @lowlydba ! 
I'll fix it when I fix #258 , before this stuff is released.

Reply:
Fixed in https://github.com/ansible-collections/community.hashi_vault/pull/264 if you have a moment @lowlydba , thank you!!